src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -100,5 +101,21 @@ private static int ComputeSymbolMatchPriority(ISymbol symbol)
 
             return SymbolMatchPriority.PreferType;
         }
+
+        protected override async Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var name = SymbolCompletionItem.GetSymbolName(item);
+            var semanticModel = await document.GetSemanticModelForSpanAsync(new TextSpan(position, 0), cancellationToken)
+                .ConfigureAwait(false);
+            var symbols = await Recommender.GetImmutableRecommendedSymbolsAtPositionAsync(
```

Comment:
instead of taking an semantic model, workspace and options, could this not just take a document?